### PR TITLE
Destination can fetch random record

### PR DESCRIPTION
**Pull request by krishnaglick**
## Change description

Removed `allowFetchWithoutRecordId` property as it was preventing destinations from properly loading a random record.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: ui/ui-components/components/destination/DestinationSampleRecord.tsx [hunk 1]
```diff
@@ -102,7 +102,6 @@ const DestinationSampleRecord: React.FC<Props & SampleRecordOmittedProps> = ({
       warning={warning}
       propertyLinkDisabled
       importDisabled
-      allowFetchWithoutRecordId
```
Comment (krishnaglick) [inline ui/ui-components/components/destination/DestinationSampleRecord.tsx:105]: @edmundito This didn't _seem_ to break anything.

File: ui/ui-components/components/record/SampleRecordCard.tsx [hunk 2]
```diff
@@ -86,7 +85,6 @@ const SampleRecordCard: React.FC<SampleRecordCardProps> = ({
   highlightProperty,
   highlightPropertyError,
   propertyLinkDisabled = false,
-  allowFetchWithoutRecordId = false,
```
Comment (krishnaglick) [inline ui/ui-components/components/record/SampleRecordCard.tsx:89]: This was always `false` after the other change.

**Review by pedroslopez — APPROVED** [review]
This seems to work ok for me, though I'll defer to @edmundito on why `allowFetchWithoutRecordId` was there in the first place, there might be some edge case I'm not seeing

**Comment by edmundito** [comment]
Did you test these changes against creating a new model in UI enterprise? See if there are any bad network calls being made?

**Comment by krishnaglick** [comment]
> Did you test these changes against creating a new model in UI enterprise? See if there are any bad network calls being made?

I didn't see any errors popping up during my tests.

**Review by edmundito — APPROVED** [review]

**Review by edmundito — APPROVED** [review]
